plugins/woocommerce/includes/wc-product-functions.php

```diff
@@ -686,7 +686,7 @@ function wc_get_product_id_by_sku( $sku ) {
  */
 function wc_get_product_variation_attributes( $variation_id ) {
 	// Build variation data from meta.
-	$all_meta                = get_post_meta( $variation_id );
+	$all_meta                = array_filter( (array) get_post_meta( $variation_id ) );
```

Comment:
we should change this to use the `is_array` check, there may be cases where an empty value for a meta is meaningful information

Reply:
@vedanshujain thanks for reviewing my PR. :pray: 

It makes sense to me and I've updated the code.